src/gestures.c

```diff
@@ -647,14 +631,20 @@ static int trigger_swipe_unsafe(struct Gestures* gs,
 	}
 	if (gs->move_type != move_type_to_trigger){
 		trigger_delayed_button_uncond(gs);
-		gs->move_dist = 0;
+		gs->move_dist = ABSVAL(dist);
 	}
-	else if (gs->move_dir != dir){
-		gs->move_dist = 0;
+	else if (gs->move_dir != dir && dir != TR_NONE){
+		// Reset move_dist since we're now swiping in a different direction.
+		// Don't reset if we simply don't know what direction we're going in.
+		// NOTE: That this means you can "wiggle" your fingers to build up
+		// move_dist without ever moving your fingers! The fix is a major
+		// redesign to how swiping works.
+		gs->move_dist = ABSVAL(dist);
```

Comment:
Is it safe to not set move_dist to 0 here?
I didn't noticed any bugs correlated to this, but I'm not sure why move_dist shouldn't be 0 when direction of movement is uncertain. 
Isn't it OK to ignore some vague movements?
Why it shouldn't be 0 here?

btw. c++ style comments are rarely used in this project. Please use C style instead.

Reply:
In general all the changes from int to float are related to this point so I'll just say here.

For example, when configuring mtrack to do a 3 finger drag, it is only possible to configure mtrack to start the drag gesture after at least 1px of movement. This is fine so that can stay.

However float to int conversions are being done, causing mtrack to believe at each frame that the user is not moving the cursor. The end result is that when you just begin doing a 3 finger drag, the first few frames where the cursor is moving slowly is discounted, and the 3 finger drag starts on the wrong position (accumulated errors due to int trunctation).

I made all these fixes a long time ago so I basically don't remember the exact details anymore, but that's the gist of it.